src/app/features/unit/unit-form/unit-form.component.ts

```diff
@@ -45,15 +48,23 @@ export class UnitFormComponent implements OnInit, OnDestroy {
 
   onSubmit(): void {
     this.unitForm.markAllAsTouched();
-    ///todo in next task: validate if unitName exist, send form and userId to backend, navigate to confirmation page
-    /// link to next task https://github.com/akademia-envelo-3/MELO-front/issues/51
-    sessionStorage.clear();
+    if (this.unitForm.valid) {
+      sessionStorage.clear();
+      this.unitFormService.sendNewUnitData(
+        this.unitForm.value.unitName!,
+        this.unitForm.value.unitDescription!
```

Comment:
Zamiast tego powinno wystarczyć ``this.unitForm.getRawValue()`` aby ts nie krzyczał błędami.

Reply:
poprawione